### PR TITLE
network api base

### PR DESCRIPTION
**Pull request by vladikr**
Signed-off-by: Vladik Romanovsky <vromanso@redhat.com>

### CODE REVIEW

**Comment by vladikr** [comment]
kubevirt/kubevirt/#1075

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Interfaces and Networks
+
+An `interface` defines a virtual network interface of a virtual machine.
+
+A `network` specifies the backend of an `interface` and controlls to which logical or physical device it is connected to.
```
Comment (booxter) [inline workloads/virtual-machines/interfaces-and-networks.md]: -> controls

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+## Customizing network interfaces and networks
+
+In order customize the network connectivity of a virtual machine, users must provide
+
+- a list of interfaces and
+- a list of networks each of the interfaces is connected to
+
+The `name` property of an interface and network items defines which interface is connected to which network.
+This mean sthat if an interface is named `red`, then it will be connected to a network which is also named `red`:
```
Comment (booxter) [inline workloads/virtual-machines/interfaces-and-networks.md]: mean sthat -> means that

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+### `pod` backend
+
+|Feature||
+|--|--|
+| Supported protocols | TCP, UDP |
+| Eventually supported protocols | IP |
+
+This backend type is used if an interface should be connected to the regular pod network interface.
+
+In some cases the underlying network plugin (flannel, weave, OpenShift SDN) acts as an ethernet bridge or switch, in those cases the `pod` backend can also be used to provide an IP level connectivity to an interface (see backend `bridge.delegateIP`).
```
Comment (booxter) [inline workloads/virtual-machines/interfaces-and-networks.md]: -> Ethernet (upper case), here and below

File: workloads/virtual-machines/interfaces-and-networks.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+| Connection method | Description |
+|--|--|
+| `bridge` | Bridge the interface to the network |
+
+### `bridge` connection
+
+This connection will create an ethernet bridge between the interface and the network.
+
+If the `delegateIp` option is used, then - if available - an IP address assigned to the network device representing the network, will be offered to the virtual machine via DHCP and be removed from the interface itself. The effect is that the IP endpoitn is "moved" form the network interface to the virtual machine.
```
Comment (booxter) [inline workloads/virtual-machines/interfaces-and-networks.md]: endpoitn -> endpoint

File: workloads/virtual-machines/interfaces-and-networks.md (hunk 1 above)
Comment (fabiand) [reply workloads/virtual-machines/interfaces-and-networks.md]: Done

File: workloads/virtual-machines/interfaces-and-networks.md (hunk 2 above)
Comment (fabiand) [reply workloads/virtual-machines/interfaces-and-networks.md]: Done

File: workloads/virtual-machines/interfaces-and-networks.md (hunk 3 above)
Comment (fabiand) [reply workloads/virtual-machines/interfaces-and-networks.md]: Done

File: workloads/virtual-machines/interfaces-and-networks.md (hunk 4 above)
Comment (fabiand) [reply workloads/virtual-machines/interfaces-and-networks.md]: Done

**Pull request merged**